src/server/hset_family.cc

```diff
@@ -1097,7 +1097,9 @@ void HSetFamily::HRandField(CmdArgList args, ConnectionContext* cntx) {
       }
 
       if (string_map->Empty()) {
-        db_slice.Del(db_context.db_index, *it_res);
+        auto it_mutable = db_slice.FindMutable(db_context, key, OBJ_HASH);
```

Comment:
It's the weirdness of our field expiration feature..
In Redis, an empty set is non-existing. Here, if we discover that all fields have expired, we remove the key. Yes, it could happen in a readonly command :(

We should do it in all commands I guess, or better yet, implement it as a similar mechanism to expiration. Otherwise, an attempt to, say, set a string to what was previously a hash key will fail, despite it being empty.

I'd say this is of low priority though (and also outside the scope here)